### ISSUE TITLE
Request a VSA spot check review for Public Websites Yellow Ribbon

### ISSUE THREAD

**Issue by MickinSahni** [department-of-veterans-affairs/va.gov-team#7344]
### For reference

**VSA Development Review** - [https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/teams/vsa/accessibility/vsa-accessibility-review-process.md#development-review](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/teams/vsa/accessibility/vsa-accessibility-review-process.md#development-review)

## Team Name

Public Websites

## Product/Feature Name

Yellow Ribbon

## Product description

> To provide Yellow Ribbon. (YR) participating school information as more user-friendly content than the current experience of 50+ manual table pages. The current experience exists as it does today because there is no single source of Yellow Ribbon data.
> 
> Our MVP will be a single-page application that leverages an existing API (GI Data Service) and allows users to search for YR schools based on relevant criteria.

- [ ] **Development intent review** (Review design to discuss approach to take with development, identify what components and semantic HTML would be appropriate, identify any accessibility issues and how to address) - This would happen as designs are passed to development, before development begins.

- [x] **Spot check review** (Review and follow the steps in the [508 Checklist](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/508-checklist.md) first.) This happens well before launch, so that the 508 office can review and any issues addressed *before* launch.

- [ ] **Other:** Please indicate what you're requesting

## Point of contact

Mickin

## Preferred times for dev intent checkpoint

n/a

## Staging URL and credentials

https://staging.va.gov/education/yellow-ribbon-participating-schools/

## Test plan

https://github.com/department-of-veterans-affairs/va.gov-team/issues/6918

## Previous feedback/reviews
* https://github.com/department-of-veterans-affairs/va.gov-team/issues/5402
* https://github.com/department-of-veterans-affairs/va.gov-team/issues/5324
* https://github.com/department-of-veterans-affairs/va.gov-team/issues/4944
* https://github.com/department-of-veterans-affairs/va.gov-team/issues/4470

**Comment by jenstrickland** [comment]
@MickinSahni I don't see any test cases in TestRail. My guess is we want to test:

1. A search that returns no results
1. A search that returns a single result (I imagine that's a possibility, but may be wrong)
1. A search that returns many results
1. Should we test for a server error?

@tlei123 — thoughts?

**Comment by jenstrickland** [comment]
@MickinSahni Here are the results of the spot check a11y review. Let me know if anything is unclear or if there are any questions. 

1. (3x) Form elements **must** have labels | 508-issue-semantic-markup
    `<input class="usa-input" name="yr-search-name" type="text" value="">`
    - Potential fix: add id="yr-search-name" to the input
    `<select name="yr-search-state">`
    - Potential fix: add id="yr-search-state" to the select
    `<input class="usa-input" name="yr-search-city" type="text" value="">`
    - Potential fix: add id="yr-search-city" to the select

    See https://design.va.gov/components/form-controls

<hr>

2. Heading levels **should** only increase by one | axe-core, screenreader
    `<h3 class="vads-u-margin-top--1p5">Search criteria</h3>`
    - Potential fix: make this an `<h2>`

<hr>

3. Focus should be managed. | 508-issue-focus-mgmt, screenreader
    As Liz mentioned, focus **should** move from the Search button to the H1 on the search results page. If a search is updated on the results screen, then focus **should** move to the results heading. 

    ![Screen Shot 2020-04-06 at 10 52 04 AM](https://user-images.githubusercontent.com/57469/78590191-73bf9780-780f-11ea-9946-3cc2cad0168f.png)

<hr>

4. Zoom to 400% at 1280px — Reveals a number of cut off text areas. | zoom, 508-issue-mobile
    These **should** not be cut off, and are likely to be addressed as the mobile design/development rolls out.
[see screenshots]
    ![Screen Shot 2020-04-06 at 9 52 30 AM](https://user-images.githubusercontent.com/57469/78590018-24796700-780f-11ea-8be1-ddad5b8e933e.png)
    ![Screen Shot 2020-04-06 at 10 50 37 AM](https://user-images.githubusercontent.com/57469/78590020-24796700-780f-11ea-8cd9-7b32f9ab612e.png)
    ![Screen Shot 2020-04-06 at 10 51 08 AM](https://user-images.githubusercontent.com/57469/78590162-630f2180-780f-11ea-95ef-4a80454cda9c.png)


<hr>

5. Focus outline **should** be evident on the pagination links. This may be a sitewide design system issue. | 508-issue-focus-mgmt

    ![Screen Shot 2020-04-06 at 10 52 15 AM (2)](https://user-images.githubusercontent.com/57469/78590291-9baefb00-780f-11ea-895a-bcc490225126.png)

<hr>

6. The search results **should** read as a list for a screen reader. | 508-issue-cognition, screenreader

    **Consider** using one of these markup patterns:
    - ol or ul
    - dl, where each school is a dt with dd paired with it
    - table 

<hr>

7. For non-sighted users, consider adding the `<dfn class="sr-only">Label of what the info is</dfn>` for each item.  | 508-issue-cognition, screenreader

<hr>

8. In the keyboard test, I got stuck on the Search button. When I tabbed, I went to the content below, and it took a long time to get to the search results, which was disorienting as a sighted, keyboard and screen reader user. | 508-issue-focus-mgmt

<hr>

9. All the "heading level 4" becomes extra noisy as a screen reader user. | 508-issue-cognition, screenreader

<hr>

## Potential fix

### Current code
```html
<div class="search-results vads-u-margin-top--2">
  <div class="medium-screen:vads-l-col vads-l-col vads-u-margin-bottom--2 vads-u-padding-x--2 vads-u-padding-y--2 vads-u-background-color--gray-light-alt vads-u-border--3px vads-u-border-color--transparent">
    <h3 class="vads-u-margin--0">Abilene Christian University</h3>
    <p class="vads-u-margin-bottom--1 vads-u-margin-top--0">Abilene, TX</p>
    <div class="vads-l-row vads-u-margin-top--2">
        <div class="vads-l-col--6 vads-u-display--flex vads-u-flex-direction--column vads-u-justify-content--space-between">
          <div class="vads-u-col">
              <h4 class="vads-u-font-family--sans vads-u-font-size--h5 vads-u-margin--0">Maximum Yellow Ribbon funding amount<br>(per student, per year)</h4>
              <p class="vads-u-margin--0">All tuition and fees not covered by Post-9/11 GI Bill benefits</p>
          </div>
          <h4 class="vads-u-font-family--sans vads-u-font-size--h5 vads-u-margin-top--2 vads-u-margin-bottom--0">Funding available for</h4>
          <p class="vads-u-margin-top--0 vads-u-margin-bottom--0">All eligible students</p>
          <h4 class="vads-u-font-family--sans vads-u-font-size--h5 vads-u-margin-top--2 vads-u-margin-bottom--0">School website</h4>
          <p class="vads-u-margin-top--0 vads-u-margin-bottom--0"><a href="www.acu.edu" rel="noreferrer noopener">www.acu.edu</a></p>
        </div>
        <div class="vads-l-col--6 vads-u-padding-left--2">
          <h4 class="vads-u-font-family--sans vads-u-font-size--h5 vads-u-margin--0">Degree type</h4>
          <p class="vads-u-margin--0">All</p>
          <h4 class="vads-u-font-family--sans vads-u-font-size--h5 vads-u-margin-top--7 vads-u-margin-bottom--0">School or program</h4>
          <p class="vads-u-margin--0">All</p>
        </div>
    </div>
  </div>

  <div class="medium-screen:vads-l-col vads-l-col vads-u-margin-bottom--2 vads-u-padding-x--2 vads-u-padding-y--2 vads-u-background-color--gray-light-alt vads-u-border--3px vads-u-border-color--transparent">
    <h3 class="vads-u-margin--0">Abraham Baldwin Agricultural College</h3>
    <p class="vads-u-margin-bottom--1 vads-u-margin-top--0">Tifton, GA</p>
    <div class="vads-l-row vads-u-margin-top--2">

<!-- etcetera -->
</div>
```

### Potential fix
```html
<dl>
    <dt role="heading" aria-level="3">
        <dfn class="sr-only">School name</dfn> 
        Abilene Christian University
    </dt>
    <dd>
        <dfn class="sr-only">City, State</dfn> 
        Abilene, <abbr title="Texas">TX</abbr>
    </dd>
    <dd>
        <dfn>Maximum Yellow Ribbon funding amount<br>(per student, per year)</dfn>
        All tuition and fees not covered by Post-9/11 GI Bill benefits
    </dd>
    <dd>
        <dfn>Funding available for</dfn>
        All eligible students</p>
    <dd>
        <dfn>School website</dfn>
        <a href="www.acu.edu" rel="noreferrer noopener">www.acu.edu</a>
    </dd>
    <dd>
        <dfn>Degree type</dfn>
        All
    </dd>
    <dd>
        <dfn>School or program</dfn>
        All
    </dd>

    <dt role="heading" aria-level="3"><dfn class="sr-only">School name</dfn> Abraham Baldwin Agricultural College</dt>
    <dd><dfn class="sr-only">City, State</dfn> Tifton, <abbr title="Georgia">GA</abbr></p>

<!-- etcetera -->
</dl> 
```

**Comment by MickinSahni** [comment]
> @MickinSahni I don't see any test cases in TestRail. My guess is we want to test:
> 
> 1. A search that returns no results (Maine, Limestone with both checkboxes checked returns no results)
> 2. A search that returns a single result (I imagine that's a possibility, but may be wrong)(Maine, Portland with both checkboxes checked returns one result)
> 3. A search that returns many results
>    
>    1. Massachusetts, Cambridge with both checkboxes checked returns 7 results
>    2. Massachusetts, Cambridge with both checkboxes unchecked returns 12 results
>    3. Florida, no city with both checkboxes unchecked returns 389 results
>    4. No state, no city with both checkboxes unchecked returns 5,569 results
> 4. Should we test for a server error?
> 
> @tlei123 — thoughts?

Appreciate this, @jenstrickland. I'm taking an action item to add those test scenarios to Tze's [ticket tracking Test Plan development](https://github.com/department-of-veterans-affairs/va.gov-team/issues/6918).

**Comment by MickinSahni** [comment]
> @MickinSahni Here are the results of the spot check a11y review. Let me know if anything is unclear or if there are any questions. https://github.com/department-of-veterans-affairs/va.gov-team/issues/7344#issuecomment-609955111

Thanks, @jenstrickland. To confirm -- there's one critical finding related to form element labels? Are there any findings that are CMS-related?

**Comment by jenstrickland** [comment]
> Thanks, @jenstrickland. To confirm -- there's one critical finding related to form element labels? Are there any findings that are CMS-related?

Is Yellow Ribbon using the CMS? 

I'm not sure if the design system is constraining some of the dev decisions here.

**Comment by jenstrickland** [comment]
@MickinSahni I just found another issue that I'd forgotten to put in my text file. I believe this is a design system, sitewide issue. 

The problem: When JAWS+IE11 (I know, cringe, when I lead with those words…) reads the checkbox, it is as shown in this screenshot. It reads the label, then the label again, and then the checkbox. When a user attempts to activate the checkbox, focus is already on the next item, so I've ended up going elsewhere, or having to shift-tab back to activate the right one. cc @1Copenut I think this is an issue we've discussed previously.

<img width="1280" alt="Screen Shot 2020-04-06 at 11 57 06 AM" src="https://user-images.githubusercontent.com/57469/78599467-306d2500-781f-11ea-84ad-afbe06b34ff2.png">

**Comment by stephorkandatx** [comment]
Going to close this out - the request was completed. We are working to address found issues
